Assert no temp file leaks

juj opened the pull request:
Make the test runner assert for each test that they don't leak temp files or directories, if environment variable EM_TESTRUNNER_DETECT_TEMPFILE_LEAKS is set.

kripken commented on tests/runner.py:
```diff
@@ -70,6 +72,11 @@ def is_emterpreter(self):
   def setUp(self):
     Settings.reset()
+
+    for root, dirnames, filenames in os.walk(TEMP_DIR):
```
i worry this will be very slow if the user has 1,000,000 files in the temp dir already. I know I sometimes do in `/tmp`...

juj replied:
Do you have a better idea on how we could ensure testing that Emscripten operations do not leak temp files or directories? Currently we still have a potential for such leaks, especially when tests fail. This is only the test runner though, can you free up /tmp regularly to free up space and make sure that this function will run fast? :)

kripken replied:
Well, if it's only the test runner leaking, i'm not too worried - we can clean the bot directories manually too, like you suggested here. But we should have tests on emcc not leaking, if we don't already.

Another worry with this pull is false positives, if another process adds a temp file meanwhile.

Overall I understand your concern, and it bugs me too. But I'm not sure what is a good solution, except to check and find leaks manually and fix them.

juj commented:
Hey, long time since progress here, dropped this for quite a while. Recently I set up a new Windows build bot, and set its emscripten temp directory point to a 8GB dedicated ram drive for this which seems to have insane speed benefits for the bot. Ensuring that it doesn't fill up in the test runner is a good idea, so picking this up again.

@kripken: addressed your review comments by performing the file leak check only if an environment variable `EM_TESTRUNNER_DETECT_TEMPFILE_LEAKS` is set. This avoids any potential performance issues. (there won't be performance issues for the bots either, since their temp directories are always dedicated for Emscripten use only, and no other programs write to them). How does this look to you now?

kripken commented:
lgtm
